idempotent-consumer/README.md

```diff
@@ -0,0 +1,72 @@
+---
+title: "Idempotent Consumer Pattern in Java: Ensuring Reliable Message Processing"
+shortTitle: Idempotent Consumer
+description: "Learn about the Idempotent Consumer pattern in Java. Discover how it ensures reliable and consistent message processing, even in cases of duplicate messages."
+category: Messaging
+language: en
+tag:
+    - Messaging
+    - Fault tolerance
+    - Event-driven
+    - Reliability
+---
+
+## Also known as
+
+* Idempotency Pattern
+
+## Intent of Idempotent Consumer Pattern
+
+The Idempotent Consumer pattern is used to handle duplicate messages in distributed systems, ensuring that multiple processing of the same message does not cause undesired side effects. This pattern guarantees that the same message can be processed repeatedly with the same outcome, which is critical in ensuring reliable communication and data consistency in systems where message duplicates are possible.
+
+## Detailed Explanation of Idempotent Consumer Pattern with Real-World Examples
+
+### Real-world Example
+
+> In a payment processing system, ensuring that payment messages are idempotent prevents duplicate transactions. For example, if a user’s payment message is accidentally processed twice, the system should recognize the second message as a duplicate and prevent it from executing a second time. By storing unique identifiers for each processed message, such as a transaction ID, the system can skip any duplicate messages. This ensures that a user is not charged twice for the same transaction, maintaining system integrity and customer satisfaction.
+
+### In Plain Words
+
+> The Idempotent Consumer pattern prevents duplicate messages from causing unintended side effects by ensuring that processing the same message multiple times results in the same outcome. This makes message processing safe in distributed systems where duplicates may occur.
+
+### Wikipedia says
+
+> In computing, idempotence is the property of certain operations in mathematics and computer science whereby they can be applied multiple times without changing the result beyond the initial application.
+
```

Comment:
We should add programmatic example section that shows minimal code snippets to implement the pattern

Reply:
Just added a programmatic example for the pattern